pygmt/src/plot.py

```diff
@@ -218,6 +219,14 @@ def plot(self, x=None, y=None, data=None, sizes=None, direction=None, **kwargs):
             )
         extra_arrays.append(sizes)
 
+    if "I" in kwargs and isinstance(kwargs["I"], np.ndarray):
```

Comment:
I think we want to allow any array-like input, not just numpy arrays. The `virtualfile_from_data` function (see #990) will be handle any array-like input automatically.

```suggestion
    if "I" in kwargs and not isinstance(kwargs["I"], str):
```

This would follow the syntax of `kwargs["G"]` above:

https://github.com/GenericMappingTools/pygmt/blob/8e2b2e6fd4900ccae25282d9d50f0ccd34d400ee/pygmt/src/plot.py#L207-L208

Reply:
I think we should use the `is_nonstr_iter` function:

https://github.com/GenericMappingTools/pygmt/blob/69417b4df78c99c178fa3b47ac4617f181ff3fa5/pygmt/helpers/utils.py#L161

Reply:
Using `is_nonstr_iter` to check the input `intensity` (3503319)